cmd/internal/planetscale_connection.go

```diff
@@ -56,8 +56,8 @@ func (psc PlanetScaleSource) GetInitialState(keyspaceOrDatabase string, shards [
 		}
 
 		// if we got this far, all the shards that the customer asked for exist in the PlanetScale database.
-		filteredShards := make([]string, len(foundShards))
-		for key := range foundShards {
+		filteredShards := make([]string, len(configuredShards))
+		for _, key := range configuredShards {
 			filteredShards = append(filteredShards, key)
 		}
```

Comment:
Is there a reason we need to do this instead of just using `configuredShards` to begin with? we're already checking that every shard in `configuredShard` exists or we're returning, so the two arrays are identical.

Reply:
I read what you wrote a few more times, you're right, we don't need this intermediate variable 👍🏾